### PR TITLE
Fixes black background for png images in wysiwyg editors

### PR DESCRIPTION
**Pull request by eduard13**
### Description
This PR fixes the black background for png images, which are uploaded in wysiwyg editors. This issue appears only for GD2 image processing library.

### Fixed Issues (if relevant)
1. magento/magento2#14248: Transparent background becomes black for thumbnails of PNG into Wysiwyg editor

### Manual testing scenarios
1. Upload a png image from any wysiwyg editor
2. The background while previewing of this image should be transparent.
61eb1dd578

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Comment by rogyar** [comment]
It looks like we should perform a check that `$this->_imageHandler` is not bool|null before passing it to `$this->_fillBackgroundColor`. That's the exact reason why we have unit tests failing here (it's a sort of unit-integration test :) )

**Comment by eduard13** [comment]
Thank you a lot @rogyar for the suggestion.

**Comment by osrecio** [comment]
@eduard13 Can you check the errors in Travis?

**Comment by eduard13** [comment]
@osrecio looks like this issue is caused by declaring strict types. 

Do you think is it ok, if we set 1 in order to pass an int value?
Or is better to remove _strict_types_ declaration? Looks like because of the strict type, instead of **int** value, receives a **bool** one.
Or maybe you have some other suggestions here?

https://github.com/eduard13/magento2/blob/48607aa5dcb308a27994f35aaf330e631d9a0c8b/lib/internal/Magento/Framework/Image/Adapter/Gd2.php#L184-L185

Thank you.

File: lib/internal/Magento/Framework/Image/Adapter/Gd2.php [hunk 1]
```diff
@@ -3,13 +3,19 @@
...
 
 /**
  * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
  */
 class Gd2 extends \Magento\Framework\Image\Adapter\AbstractAdapter
 {
+    /**
+     * Keep the transparency for the images in the wysiwyg editors
+     */
+    const KEEP_TRANSPARENCY = true;
```
Comment (sidolov) [inline lib/internal/Magento/Framework/Image/Adapter/Gd2.php]: I think we don't need constant fir true value in this class, just assign true to property when it needed

**Review by sidolov — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Image/Adapter/Gd2.php (hunk 1 above)
Comment (eduard13) [reply lib/internal/Magento/Framework/Image/Adapter/Gd2.php]: Updated it, thanks.

**Review by osrecio — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @osrecio, thank you for the review.
[ENGCOM-2855](https://jira.corp.magento.com/browse/ENGCOM-2855) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.7. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**